src/domain/model/authentication/AuthenticatorFactory.ts

```diff
@@ -0,0 +1,24 @@
+import { Authenticator } from './Authenticator';
+import { AuthenticationMethod } from '../user/AuthenticationMethod';
+import { InvalidDomainStateError } from '../InvalidDomainStateError';
+
+export class AuthenticatorFactory {
+  private authenticators: Map<AuthenticationMethod, Authenticator>;
+
+  constructor(authenticators: Authenticator[]) {
+    this.authenticators = new Map(authenticators.map((authenticator) => [authenticator.handles, authenticator]));
+    this.checkAllAuthenticationMethodsImplemented();
+  }
+
+  public authenticatorFor(method: AuthenticationMethod): Authenticator {
+    return this.authenticators.get(method)!;
```

Comment:
Haven't seen this syntax before, what's it doing?

Reply:
Asserting that it's not null